libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -350,7 +351,24 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 						"(remove the \"storage\" or \"calldata\" keyword)."
 					);
 				if (varLoc == Location::Default || !contract.isLibrary())
+				{
+					if (varLoc == Location::Default && !_variable.isEventParameter())
+					{
+						if (v050)
+							typeError(
+									_variable.location(),
+									"Location must be specified as \"memory\" "
+									"for parameters in publicly visible functions."
+							);
+						else
+							m_errorReporter.warning(
```

Comment:
Since this is listed as a breaking change, there is no need to have the non-0.5.0 mode here.